File: starlore_app/lib/models/article.dart
/// 文章摘要（列表用）
class Article {
  final int id;
  final String title;
  final String description;
  final String? coverImage;
  final String category;
  final String authorName;
  final DateTime createdAt;
  final int viewCount;
  final List<String> tags;
  final bool isPublic;

  const Article({
    required this.id,
    required this.title,
    required this.description,
    this.coverImage,
    required this.category,
    required this.authorName,
    required this.createdAt,
    this.viewCount = 0,
    this.tags = const [],
    this.isPublic = true,
  });

  factory Article.fromJson(Map<String, dynamic> json) {
    return Article(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      coverImage: json['cover_image'] ?? json['coverImage'],
      category: json['category'] ?? '未分类',
      authorName: json['authorName'] ?? json['author_name'] ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString()) ?? DateTime.now()
          : DateTime.now(),
      viewCount: json['view_count'] ?? json['viewCount'] ?? 0,
      tags: (json['tags'] as List?)?.map((e) => e.toString()).toList() ?? [],
      isPublic: json['is_public'] ?? json['isPublic'] ?? true,
    );
  }
}

/// 文章详情（含 content）
class ArticleDetail extends Article {
  final String content;

  const ArticleDetail({
    required super.id,
    required super.title,
    required super.description,
    super.coverImage,
    required super.category,
    required super.authorName,
    required super.createdAt,
    super.viewCount,
    super.tags,
    super.isPublic,
    required this.content,
  });

  factory ArticleDetail.fromJson(Map<String, dynamic> json) {
    return ArticleDetail(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      content: json['content'] ?? '',
      coverImage: json['cover_image'] ?? json['coverImage'],
      category: json['category'] ?? '未分类',
      authorName: json['authorName'] ?? json['author_name'] ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString()) ?? DateTime.now()
          : DateTime.now(),
      viewCount: json['view_count'] ?? json['viewCount'] ?? 0,
      tags: (json['tags'] as List?)?.map((e) => e.toString()).toList() ?? [],
      isPublic: json['is_public'] ?? json['isPublic'] ?? true,
    );
  }
}

File: starlore-py/app/models/article.py
"""文章模型。"""

from datetime import datetime

from sqlalchemy import Integer, String, Text, DateTime, JSON, Boolean
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Article(Base):
    __tablename__ = "articles"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(Integer)
    title: Mapped[str] = mapped_column(String(200))
    content: Mapped[str | None] = mapped_column(Text)
    description: Mapped[str | None] = mapped_column(String(500))
    category: Mapped[str | None] = mapped_column(String(50))
    tags: Mapped[list | None] = mapped_column(JSON)
    cover_image: Mapped[str | None] = mapped_column(String(500))
    view_count: Mapped[int] = mapped_column(Integer, default=0)
    status: Mapped[str] = mapped_column(String(20), default="published")
    is_public: Mapped[bool | None] = mapped_column("is_public", Boolean, default=False)
    createdAt: Mapped[datetime | None] = mapped_column(DateTime)  # noqa: N815
    updatedAt: Mapped[datetime | None] = mapped_column(DateTime)  # noqa: N815

